src/gui/mzroll/eicwidget.cpp

```diff
@@ -1994,35 +1984,39 @@ void EicWidget::timerEvent(QTimerEvent * event) {
 	}
 }
 
-/*
-@author: Sahil
-*/
-//TODO: Sahil, Added this function while merging eicwidget
-void EicWidget::addMS2Events(float mzmin, float mzmax) {
-
+void EicWidget::addMS2Events(float mzmin, float mzmax)
+{
     qDebug() << "addMS2Events() " << mzmin << " " << mzmax;
 
     MainWindow* mw = getMainWindow();
     vector <mzSample*> samples = mw->getVisibleSamples();
 
     if (samples.size() <= 0 ) return;
-    MassCutoff *massCutoff = mw->getUserMassCutoff();
-
-    mw->fragPanel->clearTree();
-    int count=0;
-    for ( unsigned int i=0; i < samples.size(); i++ ) {
-        mzSample* sample = samples[i];
-
-        for (unsigned int j=0; j < sample->scans.size(); j++ ) {
-            Scan* scan = sample->scans[j];
 
-            if (scan->mslevel > 1 && scan->precursorMz >= mzmin && scan->precursorMz <= mzmax) {
+    //clear MS2 events list
+	mw->fragPanel->clearTree();
+    
+	int count = 0;
+    for (auto sample : samples) {
+        for (auto scan : sample->scans) {
```

Comment:
We need to use `auto&` to prevent copying .
Forgot to mention this in the previous PRs

Reply:
auto const & seems more suitable then. We're not modifying the elements.

Reply:
`const auto&`

Reply:
Is that a convention? both `auto const &` and `const auto&` work the same, right?

Reply:
`const auto&` is pretty much the convention. I use `auto&` almost always (unless its a pointer variable, like in this case), but `const` is not always possible (not sure if it is in this case or now).